src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,39 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  #this;
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(animal) {
+    if (
+      !(animal instanceof Carnivore) &&
+      animal instanceof Herbivore &&
+      animal.hidden === false
+    ) {
```

Comment:
The conditional check in the 'bite' method is correct, ensuring that the method only affects 'Herbivore' instances that are not hidden and does not affect 'Carnivore' instances.